### PR TITLE
quodlibet: 2.6.3 -> 3.9.1

### PR DESCRIPTION
**Pull request by sauyon**
###### Motivation for this change

Quod Libet 2.6.3 is ancient

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] Linux
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/top-level/python-packages.nix
Comment (FRidh) [inline pkgs/top-level/python-packages.nix]: Can you move this expression to `python-modules/faulthandler/default.nix`.

File: pkgs/top-level/python-packages.nix
Comment (FRidh) [inline pkgs/top-level/python-packages.nix]: fetchPypi

File: pkgs/top-level/python-packages.nix
Comment (FRidh) [inline pkgs/top-level/python-packages.nix]: `python-modules/pyobjc/default/nix`

File: pkgs/top-level/python-packages.nix
Comment (FRidh) [inline pkgs/top-level/python-packages.nix]: fetchPypi

**Review by FRidh — CHANGES_REQUESTED** [review]
Python packages need some work, meta is missing.

**Comment by sauyon** [comment]
The license for quodlibet is stated to be

> Quod Libet and all associated modules are free software distributed entirely under the terms of the GNU GPL version 2. Alternate licensing terms (such as GNU GPL “version 2 or later”, or the GNU LGPL) may be available for certain files or modules; see the files for details.

so I put gpl2.

File: pkgs/development/python-modules/pyobjc/default.nix
Comment (FRidh) [inline pkgs/development/python-modules/pyobjc/default.nix]: `undefined variable ‘stdenv’ at /nix/store/9nl1fs575ayk3hk1bjymp03plh84wnrp-quodlibet.tar.gz/pkgs/development/python-modules/pyobjc/default.nix:16:24`

Please do test.

**Review by FRidh — CHANGES_REQUESTED** [review]

File: pkgs/development/python-modules/pyobjc/default.nix
Comment (sauyon) [reply pkgs/development/python-modules/pyobjc/default.nix]: can't, don't have a mac :/

Comment (sauyon) [reply pkgs/development/python-modules/pyobjc/default.nix]: actually I could've caught that... added licenses after building, whoops.

Comment (FRidh) [reply pkgs/development/python-modules/pyobjc/default.nix]: That's irrelevant. I got this evaluation error on NixOS. `stdenv` is not passed in.

Comment (FRidh) [reply pkgs/development/python-modules/pyobjc/default.nix]: Yea, its a common error. Happens to me as well.

**Comment by FRidh** [comment]
Something has gone wrong here with merging/rebasing. We need a single commit per package/expression.

**Comment by sauyon** [comment]
Yeah, I did the merge on github thinking "this'll be relatively easy to squash" like an idiot

**Comment by sauyon** [comment]
Do you want me to separate the python packages into their own commits?

**Comment by FRidh** [comment]
Preferably, yes.

**Comment by orivej** [comment]
@sauyon `faulthandler` should be disabled for python3, otherwise `python36Packages.faulthandler` is reachable and fails to build with `ERROR: faulthandler is a builtin module since Python 3.3`.

File: pkgs/top-level/python-packages.nix
Comment (rycee) [inline pkgs/top-level/python-packages.nix]: This line is very long, I think it would be clearer as
```
faulthandler =
  if ! isPy3k
  then callPackage ../development/python-modules/faulthandler {}
  else throw "faulthandler is built into ${python.executable}";
```

**Comment by rycee** [comment]
When attempting to build this on my NixOS box I get a problem with pyobjc:
```console
rycee@beta:~/devel/nixpkgs$ nix-build -A python2Packages.pyobjc
these derivations will be built:
  /nix/store/p7w6vw0a4p7drshch389vsj9w1ggzf2z-python2.7-pyobjc-4.0b1.drv
building path(s) ‘/nix/store/jc240j31v04spa5mv8cq2c1gkn44cr32-python2.7-pyobjc-4.0b1’
unpacking sources
unpacking source archive /nix/store/30yzmj23wqhdgp41h92qs5kjd6s6i00m-pyobjc-4.0b1.tar.gz
source root is pyobjc-4.0b1
setting SOURCE_DATE_EPOCH to timestamp 1503862173 of file pyobjc-4.0b1/setup.cfg
patching sources
configuring
building
ERROR: Requires macOS to install or build
builder for ‘/nix/store/p7w6vw0a4p7drshch389vsj9w1ggzf2z-python2.7-pyobjc-4.0b1.drv’ failed with exit code 1
error: build of ‘/nix/store/p7w6vw0a4p7drshch389vsj9w1ggzf2z-python2.7-pyobjc-4.0b1.drv’ failed
rycee@beta (✗ 100):~/devel/nixpkgs$ 
```

**Comment by sauyon** [comment]
I've reformatted and disabled pyobjc for non-darwin (it's only for mac os)

File: pkgs/top-level/python-packages.nix
Comment (rycee) [inline pkgs/top-level/python-packages.nix]: I think your rebase got mixed up. There is already a
```
pyodbc = callPackage ../development/python-modules/pyodbc { };
```
below. This results in an evaluation error:
```
error: attribute ‘pyodbc’ at /home/rycee/devel/nixpkgs/pkgs/top-level/python-packages.nix:16599:3 already defined at /home/rycee/devel/nixpkgs/pkgs/top-level/python-packages.nix:16579:3
```

**Comment by sauyon** [comment]
@rycee should be fixed.

**Comment by rycee** [comment]
Ok, looks good to me now but you must remove the unrelated "php: fix ldap paths when header exists on host system" commit before a merge can happen.

**Comment by sauyon** [comment]
@rycee odd, should be fixed.

**Comment by rycee** [comment]
Ok, now I could do a nox review and it looks good for me on NixOS. I'm a bit concerned about the error on the mac Travis though: https://travis-ci.org/NixOS/nixpkgs/jobs/297566861#L1081-L1082 Is this something that needs to be resolved? It seems limited to Python 3.6, which quodlibet does not use…

**Comment by sauyon** [comment]
@rycee it might make sense to disable it for python3, yeah. I don't know anything about it; I don't use a mac and am not the maintainer of the project.

**Comment by rycee** [comment]
Rebased to master in 7486ae8e7ccc256873982c60445f34dfa72382e9. Thanks!

**Pull request closed without merging**
